src/OrchardCore/OrchardCore.Abstractions/Shell/Builders/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -1,14 +1,66 @@
 using System;
+using System.Linq;
 using System.Threading.Tasks;
+using Microsoft.Extensions.Options;
 using OrchardCore.Environment.Shell.Builders;
 
 namespace Microsoft.Extensions.DependencyInjection;
 
 public static class ServiceCollectionExtensions
 {
     /// <summary>
-    /// Registers a delegate to be invoked asynchronously after a tenant container is created.
+    /// Registers a delegate to be invoked asynchronously just after a tenant container is created.
     /// </summary>
-    public static IServiceCollection Initialize(this IServiceCollection services, Func<IServiceProvider, ValueTask> _initializeAsync)
-        => services.Configure<ShellContainerOptions>(options => options.Initializers.Add(_initializeAsync));
+    public static IServiceCollection Initialize(this IServiceCollection services, Func<IServiceProvider, ValueTask> initializeAsync)
+        => services.Configure<ShellContainerOptions>(options => options.Initializers.Add(initializeAsync));
+
+    /// <summary>
+    /// Registers a delegate used to configure asynchronously a type of options just after a tenant container is created.
+    /// </summary>
+    public static IServiceCollection Configure<TOptions>(
+        this IServiceCollection services, Func<TOptions, ValueTask> configureAsync)
+        where TOptions : class, IAsyncOptions, new()
+    {
+        var descriptor = services.FirstOrDefault(d => d.ServiceType == typeof(TOptions));
+        if (descriptor is null)
+        {
```

Comment:
Maybe use .Any

Reply:
Quite identical but I prefer too because we don't use the descriptor instance, will change